region/maridia/inner-yellow/Kassiuz Room.json

```diff
@@ -111,15 +111,85 @@
       "requires": [
         {"resetRoom": {"nodes": [1]}},
         {"or": [
-          "Spazer",
-          "Wave",
-          "Plasma",
-          "canUseGrapple",
-          "ScrewAttack"
+          {"and": [
+            "Grapple",
+            {"cycleFrames": 180}
+          ]},
+          {"and": [
+            {"or": [
+              "Plasma",
+              "Wave",
+              "Spazer",
+              "ScrewAttack"
+            ]},
+            {"cycleFrames": 210}
+          ]},
+          {"and": [
+            "canFarmWhileShooting",
+            {"cycleFrames": 330}
+          ]},
+          {"cycleFrames": 400}
+        ]}
+      ],
+      "farmCycleDrops": [{"enemy": "Puyo", "count": 2}]
+    },
+    {
+      "link": [1, 1],
+      "name": "Puyo and Choot Farm",
+      "requires": [
+        {"or": [
+          {"resetRoom": {"nodes": [1]}},
+          {"and": [
+            {"resetRoom": {"nodes": [2]}},
+            {"cycleFrames": 120}
+          ]}
         ]},
-        {"partialRefill": {"type": "Energy", "limit": 200}},
-        {"partialRefill": {"type": "Missile", "limit": 12}},
-        {"partialRefill": {"type": "PowerBomb", "limit": 6}}
+        {"or": [
+          "canWalljump",
+          {"and": [
+            "SpaceJump",
+            {"cycleFrames": 300}
+          ]},
+          {"and": [
+            "canTrickySpringBallJump",
+            {"cycleFrames": 520},
```

Comment:
A little bit high

Reply:
Fair enough, this was the last case I checked and I was getting tired of the room, haha. Tightened it now.